src/editor/editor.js

```diff
@@ -45,7 +47,7 @@ export default class Editor {
 		 * @readonly
 		 * @member {module:core/plugin~PluginCollection}
 		 */
-		this.plugins = new PluginCollection( this );
+		this.plugins = new PluginCollection( this, availablePlugins );
```

Comment:
This is overly complicated. All we need here is `new PluginCollection( this, this.constructor.build && this.constructor.build.plugins )`

Or, at least remove the `findAvailablePlugins` function and define `availablePlugins` as:

```js
const availablePlugins = this.constructor.build && this.constructor.build.plugins;
```